clients/src/main/java/org/apache/kafka/clients/consumer/internals/Fetch.java

```diff
@@ -57,6 +75,22 @@ private Fetch(
         this.records = records;
         this.positionAdvanced = positionAdvanced;
         this.numRecords = numRecords;
+        this.nextFetchOffsets = new HashMap<>();
```

Comment:
Same here, I would be interested in when this constructor is called. I'm worried that we forget to pass the correct nextOffset information if this constructor is called.